src/screens/components/TokenOperations/index.tsx

```diff
@@ -99,35 +99,68 @@ const TokenOperations = ({
     }
   };
 
-  const submitTransaction = async () => {
-    const tx = async () => {
-      const txRequest = await getTxRequest();
-      const waiter = await waitForSeqno(
-        client.openWalletFromAddress({
-          source: Address.parse(address!!),
-        })
-      );
-      let deepLinkUrl = await walletService.requestTransaction(adapterId!!, session, txRequest);
-      if (typeof deepLinkUrl == "string") {
-        if (isMobile) {
-          window.location.href = deepLinkUrl;
-        } else {
-          setKeeperTransactionLink(deepLinkUrl);
+  //create hook for submitting transactions}
+  //check adapter type
+  //according to adapter type use exact requestTransaction
+  //hook returns status (pending, error, success)
+
+  const useSubmitTransaction = () => {
```

Comment:
usually hooks declaration is outside the component lifecycle, 
so or just make it a function and not a hook, or get the hook out of the lifecycle